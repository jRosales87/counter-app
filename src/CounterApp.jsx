import { useState } from 'react'

import PropTypes from 'prop-types'
export const CounterApp = ( { value } ) => {
    
    const [ counter, setCounter ] = useState(value)
    
    const contador = () => {
    //    setCounter(counter + 1);
        setCounter( (c) => c + 1 )
    }
    const restar = () => {
       setCounter(counter - 1);
        // setCounter( (c) => c - 1 )
    }
    const resetear = () => {
    //    setCounter(counter + 1);
        setCounter( value )
    }
    
    return (
        <>
            <h1>CounterApp</h1>
            <p>{ counter }</p>


            <button onClick={ contador }>
                +1
            </button>
            <button onClick={ restar }>
                -1
            </button>
            <button onClick={ resetear }>
                Reset
            </button>
        </>
      )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value: 1
}