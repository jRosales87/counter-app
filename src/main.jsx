


import React from 'react';
import ReactDOM from "react-dom/client";


// import { FirstApp } from './FirstApp'
// import { HelloWorldApp } from './HelloWorldApp';
import { CounterApp } from './CounterApp';


import './styles.css';




ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        {/* <FirstApp title="Hola soy Goku"  /> */}
        <CounterApp value={1} />
    </React.StrictMode>
);