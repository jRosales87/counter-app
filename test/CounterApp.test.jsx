import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('Prueba de CounterApp', () => { 
    const value = 1;
    test('Debe de hacer match con el snapshot', () => { 
        const { container } =  render( <CounterApp value={value} /> );
        expect( container ).toMatchSnapshot();
     });
     test('Debe de inicializarse en 1', () => { 
        render( 
            <CounterApp value={value}/> 
        );
        // screen.debug()
        expect( screen.getByText(value) ).toBeTruthy();
      })
      test('Debe de incrementar con el botón +1', () => { 
            render( 
                <CounterApp value={value}/> 
            );
            fireEvent.click( screen.getByText('+1') );
            expect( screen.getByText('2') ).toBeTruthy();
       });
      test('Debe de disminuir con el botón -1', () => { 
            render( <CounterApp value={value}/> );
            fireEvent.click( screen.getByText('-1') );
            expect( screen.getByText('0') ).toBeTruthy();
       });
       test('Debe de funcionar el botón de reset', () => { 

            render( <CounterApp value={value}/> );
            fireEvent.click( screen.getByText('+1') );
            fireEvent.click( screen.getByText('+1') );
            fireEvent.click( screen.getByText('+1') );
            fireEvent.click( screen.getByText('Reset') );
            expect( screen.getByText(value) ).toBeTruthy();
        });
 })