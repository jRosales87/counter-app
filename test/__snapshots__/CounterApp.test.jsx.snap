// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de CounterApp Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <p>
    1
  </p>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
